### ISSUE TITLE
Collections with times or dates not working?

### ISSUE THREAD

**Issue by bengtfalke** [darktable-org/darktable#11582]
Trying the new way to select a couple of dates. Date taken does not show any photos. Not any interval as I can see.
I would prefer Time category to work the same way as folders but with the added possibility to select a group of folder either with ctrl click or shift click as normal in most applications.

![filter_module](https://user-images.githubusercontent.com/78697169/164449218-b6c291db-cb11-496a-a8da-4d1eb56ac9f6.png)

Platform
darktable version : 3.9.0+1244~gfe1008608-dirty
OS : Win10 19042.1288
Memory : 32 GB
Graphics card : Gigabyte GeForce RTX 2070 SUPER OC 3X
Graphics driver : 30.0.14.9613
OpenCL installed : yes
OpenCL activated : yes

**Comment by Nilvus** [comment]
Can't reproduce (tested with 3.9.0+1210 and 3.9.0+1251~g2a388c0db after compiling again last master). Could you check what you have in new filters (filters collections module and top quick filters header bar). Could be a little disturbing with 3 places for filters but it's temporary (and all are synchronized together). Plan is to move completely to new filters in 4.2. You probably have so another filter set. You could reset all new filters by double-click on it.

**Comment by bengtfalke** [comment]
I tried it all... can you please be a bit more precise in what you want me to test. What ever I do I cant get photos selected by date-taken to show up in Lighttable. Folders works as it should. Where exactly do I dubble-click to reset all new filters?

**Comment by Nilvus** [comment]
I talk about filter collections module (just below collections one) and filters on header toolbar. See screenshots I posted in my last update of my actual work in progress PR (on your side, you will have "before" screenshots) to understand what I'm talking about: https://github.com/darktable-org/darktable/pull/11559#issuecomment-1105109088

On all widgets, you have the range to select by click & drag values (stars, or image depending on settings.. Just double-click. On my PR, I have improve visibility about selected or not parts on those new filters. Hope it's clearer now.

**Comment by bengtfalke** [comment]
Will the old Collections be replaced or complimented by Collection filters?  I have tried to reset the Collection filters and the top Filter and use just Collection + date-taken and still it shows no photos. With folders everything works as expected.

**Comment by Nilvus** [comment]
> Will the old Collections be replaced or complimented by Collection filters? I have tried to reset the Collection filters and the top Filter and use just Collection + date-taken and still it shows no photos. With folders everything works as expected.

Will be replaced but not immediately. There's some a lot of filters to add to new filters collections module (and it needs a lot of work to do). So should be replaced for 4.2, not 4.0.

I don't understand so why this don't work on your side. Could you post some screenshots of what you have on filter collections modules and on quick filters in header bar?

**Comment by bengtfalke** [comment]
I have tried to do some screen dumps to show the settings.
![collection_filter](https://user-images.githubusercontent.com/78697169/164708041-9b09d3dd-3cc9-44b7-b3d6-cf9895961ee9.png)
![date_taken](https://user-images.githubusercontent.com/78697169/164708055-783c8563-3ae7-4e88-98be-2a42132e4d48.png)

The date picker in the Collection filters is not that self-explanatory and I do get odd results sometimes with the starting date resetting to year 0001 and as I can not enter a new year in that field it is very difficult to set the min year. See below.
![date_picker](https://user-images.githubusercontent.com/78697169/164708194-e959236a-0336-44ff-9e6c-dee93dc09eee.png)

**Comment by bengtfalke** [comment]
I should be able to show photos by the date-taken in the collections with the default settings in Collection filters... shouldn't I?

**Comment by Nilvus** [comment]
Of course, you should and here, I don't understand why you don't have it. I will let see if someone else understand why or have any other tests to do. What you try works on my side. Maybe last thing, could you test without your darktablerc file? Go on your config folder and remove it (save it before of course). Be sure to do that with darktable closed. Then test again. Not sure but a test to do.

About that and last issue you post here (reset to year 0001 in date-time taken new filters, I ping @AlicVB

**Comment by bengtfalke** [comment]
I found two darktablerc files on my system and I renamed both but with no effect on this issue.

**Comment by AlicVB** [comment]
Seems that there's multiple things here... But let's take them one by one ;)
I effectively can reproduce somehow an issue : in collection module, select date-time taken, then try to select a range with ctr-click. The selection in the treeview is effective, but not "validated".
That's an issue, and I'll fix it ASAP.

What is strange, that said, is that on your screenshot the "entry" next to the rule property is filled with a "range" text (`[2022:01;2022:04]` and that doesn't happen here...

btw, do you have "single-click" option activated for collection module in the prefs ?

**Comment by bengtfalke** [comment]
Yes, I have single-click activated. Is that a problem?

**Comment by AlicVB** [comment]
no, that's not. but then the issue is a little bit different... I'll have to check that another time !

**Comment by AlicVB** [comment]
So, I've a fix for the case where the pref is not set to "single-click" but that seems a completely different issue !

... and I can't reproduce your issue with master...
So I will need your help to do some remote debugging if you are ok ! 
1. first, select the date range with shift-click (it's your issue : no images shown)
2. click inside the entry where it's written `[......;......]` and press enter => does it change anything ?
3. if not, try to click on a single entry (no range selection) => does it show some images ?
4. if not repeat the same with a "final" node (one you can't expand further) => is that ok ?

if any cases, as you seems to compile from source, can you add `printf("\n\n%s\n\n", query);` in src/common/collection.c line 148 and report what is written in terminal when you do a range selection ?

Thanks in advance !

**Comment by bengtfalke** [comment]
I am not sure if you refer to Collections or Collection Filters. I assume collection, correct me if I am wrong?
I have not compiled myself yet, so this is the Nightly builds:
 https://github.com/darktable-org/darktable/actions/workflows/nightly.yml
1.     first, select the date range with shift-click (it's your issue : no images shown):  OK
2.     click inside the entry where it's written [......;......] and press enter => does it change anything ?: No
3.     if not, try to click on a single entry (no range selection) => does it show some images ?: No
4.     if not repeat the same with a "final" node (one you can't expand further) => is that ok ? No

**Comment by AlicVB** [comment]
yes, that was the collection module. I assume from your screenshots that the collection filters are reset, so it shouldn't have any impact.
So I am right that in fact nothing is working with the "date/time" property in the collection module ? That means you can't even select something by clicking on the item.

As you are not compiling from source, can you launch dt with `darktable -d sql` in a terminal and report the output for the following steps (avoid as much as possible any other interactions to simplify the log) : 
- open dt
- select a date range in collection module
- close dt

please also report the date range you have selected (like written in the top entry, something like [2022:01;2022:04])

last, can you test if you experiment same thing with all date property, and with exposure, aspect ratio...

Thanks a lot, and sorry to annoy you but I need you to understand what happen here !

**Comment by bengtfalke** [comment]
[darktable-log.txt](https://github.com/darktable-org/darktable/files/8549955/darktable-log.txt)

None of the date property's are working but exposure, aspect ratio works as expected.

**Comment by AlicVB** [comment]
@phweyland : I fear this one is for you...
In the above log, you can see that the select query contains `(datetime_taken >= 1796292608) AND (datetime_taken <= -2093417448)` which is obviously wrong and should be something like `(datetime_taken >= 63745056000000000) AND (datetime_taken <= 63808127999999000)` (that's what I get here when I enter `[2021;2022]` in the entry in the collection module)

@bengtfalke : I've pinged our date guru :)

**Comment by bengtfalke** [comment]
@AlicVB Thank you!

**Comment by phweyland** [comment]
> I fear this one is for you...

I'll look at it !

**Comment by phweyland** [comment]
Humm... I haven't reproduced the issue.
Talking only about the input range in collection module, this seems to work for me:
![image](https://user-images.githubusercontent.com/23012047/165046987-b1ff3cc9-ba0e-48af-843c-0f267f5a7327.png)
![image](https://user-images.githubusercontent.com/23012047/165047078-377c8a57-5dee-4f37-803d-36888c0a1e64.png)
![image](https://user-images.githubusercontent.com/23012047/165047198-383ae552-1d08-40cb-8ada-f4c685eacf7c.png)

But, probably an other point, if I select (shift-click) several values in the list, the range doesn't appear in the entry. 


> Date range 2022 2021

Is that a tipo ? Should be [2021;2022] ...

**Comment by bengtfalke** [comment]
First the typo... I selected the years in that order because my years are sorted that way in the collection view but of course darktable will reverse that order i assume (i just tried to select them in the other order with the same negative result) so it should read 2021 2022.

Next it seems you are using another version than I am. In your screen dumps I do not see the Collection filter module and as I assume they are working together that might be of importance. I use the nightly builds for Windows 10 and are now on version: 3.9.0+1263~ga52359dfe-dirty of darktable. The nightly builds are here: https://github.com/darktable-org/darktable/actions/workflows/nightly.yml

**Comment by phweyland** [comment]
Well, if you invert the dates in the entry it cannot work :+1: 
`((datetime_taken >= 63776592000000000) AND (datetime_taken <= 63776591999999000))) `

This said, I don't get negative values. Would there be a windows effect here ?

**Comment by parafin** [comment]
> This said, I don't get negative values. Would there be a windows effect here ?

https://github.com/darktable-org/darktable/blob/feaa9bc409b3410adcc32767080c936b172b7e16/src/common/collection.c#L1833

long on Windows is 32-bit, while elsewhere is 64-bit (on 64 bit platform). Basically all uses of long type is not compatible and should be replaced with smth like int64_t.

**Comment by AlicVB** [comment]
@phweyland : not only he gets negative value, but also really smaller ones... for me, it's a little bit like if we wanted to cast the value to a too "small" type.... But I've not found where can be the problem :(

**Comment by phweyland** [comment]
> ong on Windows is 32-bit, while elsewhere is 64-bit (on 64 bit platform). Basically all uses of long type is not compatible and should be replaced with smth like int64_t.

Ouch ...  Thanks! In that case it may even be uint64_t

**Comment by zaha** [comment]
Sorry to exhume this thread, but I found it via a search for my problem and the issues are very similar if not the same:
I am unable to filter by date by either date picker or by entering a date and/or time, it's always setting the value to 01:01:01. The only way to filter by date is by using the timeline with the mouse, but I'm unable to enter any different values then.

I've recorded a gif to show the issue:
![2023-07-06_darktable_KjgvpKu6SV](https://github.com/darktable-org/darktable/assets/1721211/41654672-4df2-4004-8f5b-5d685b262d4f)

Platform
darktable version : 4.4.1
OS : Win11 22H2 (22621.1848)
Memory : 32 GB
Graphics card : NVIDIA GeForce RTX 3060 Laptop GPU
Graphics driver: 531.79

Is this a regression or  should I post this as a new bug?
